### PR TITLE
fix: remove paragon from bundle

### PR DESCRIPTION
**Pull request by douglashall**
This reduces the package size by removing paragon from the bundle. Consumers of this package are expected to provide paragon.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,9 +86,11 @@
     "webpack-merge": "^4.2.1"
   },
   "dependencies": {
-    "@edx/paragon": "^3.8.0",
+    "query-string": "^5.1.1"
+  },
+  "peerDependencies": {
+    "@edx/paragon": "^4.2.4",
     "prop-types": "^15.5.10",
-    "query-string": "^5.1.1",
```
Comment (robrap) [inline package.json]: Was query-string meant to be a peer dependency?  Could this have stayed and also moved from `dependencies` to `devDependencies`?

File: package.json [hunk 2]
```diff
@@ -3,7 +3,6 @@
   "version": "1.0.0-semantically-released",
   "description": "Site footer component for use when building edX frontend applications",
   "main": "dist/main.js",
-  "module": "dist/main.js",
```
Comment (robrap) [inline package.json]: Is this something we should look for in our other libraries?  Did you already do that?

File: package.json (hunk 1 above)
Comment (douglashall) [reply package.json]: The diff is a bit wonky here. In the current version, these are all dependencies. This commit moves everything, but `query-string` to peerDependencies.

File: package.json (hunk 2 above)
Comment (douglashall) [reply package.json]: I'm going to revert this. I don't think it was involved in making the bundle smaller.

**Pull request merged**